### DISCUSSION TITLE
DNS Setting as part of Advanced Settings (with ability to block Proxmox override on LXC startup)

### DISCUSSION THREAD

**Discussion by wleelavivat** [tteck/Proxmox#801] [Ideas]
Hello,

When I was trying to use the script to spin up a container I always run into DNS look up issue. Upon investigating this issue, I have found that it is due to my network configuration and the DNS default setting in the scripts.

Currently, I have my Proxmox host on one VLAN and containers/VMs on another to isolate the network. However, by default, when container is created or started up, it will override the DNS setting and use host setting. This wouldn't work since there would be cross VLAN talk and block by default. (This might be Proxmox bug since the network setting is DHCP and it should pick those setting instead of statically set it to host setting but there is a customization to work around the issue)

My suggestion would be to add the ability to define/customize the DNS setting during set up and add it to ignore the setting to ignore list on container startup (by creating an empty file, `/etc/.pve-ignore.resolv.conf`).

Here is some source:
- [What is done at container start up](https://pve.proxmox.com/wiki/Linux_Container#_guest_operating_system_configuration)
- To ignore DNS setting in container, we have to create `/etc/.pve-ignore.resolv.conf` as empty file based on the above
- Some people who ran into this issue in general [Reddit 1](https://www.reddit.com/r/Proxmox/comments/pddvyo/lxc_containers_not_receiving_dhcp_dns_details_on/), [Proxmox Forum 1](https://forum.proxmox.com/threads/how-to-disable-auto-updates-of-etc-hostsname-etc-resolv-conf-maybe-others.76186/), [Promox Forum 2](https://forum.proxmox.com/threads/how-to-disable-auto-updates-of-etc-hostsname-etc-resolv-conf-maybe-others.76186/)

Thanks!

**Comment by tteck** [comment]
> nameserver: < string >
searchdomain: < string >
Sets DNS server IP address for a container. Create will automatically use the setting from the host if you neither set searchdomain nor nameserver.


🤔

**Comment by tteck** [comment]
Would the option to set the DNS Server address solve your issues?

**Comment by tteck** [comment]
```
    ____  __________  _______    _   __
   / __ \/ ____/ __ )/  _/   |  / | / /
  / / / / __/ / __  |/ // /| | /  |/ / 
 / /_/ / /_v4/ /_/ // // ___ |/ /|  /  
/_____/_____/_____/___/_/  |_/_/ |_/   

Using Advanced Settings
Using Container Type: 1
Using Root Password: Automatic Login
Using Container ID: 112
Using Hostname: debian
Using Disk Size: 2
Allocated Cores: 1
Allocated RAM: 512
Using Bridge: vmbr0
Using IP Address: 192.168.86.252/24
Using Gateway IP Address: 192.168.86.1
👉Using DNS Search Domain: test.local
👉Using DNS Server IP Address: 192.168.86.107
Using MAC Address: Default
Using Vlan: Default
Enable Root SSH Access: Yes
Creating a Debian LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using local-lvm for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 112 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.252 
 ✓ Internet Connected
 ✓ DNS Resolved github.com to 140.82.121.3
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Customized Container
 ✓ Cleaned
 ✓ Completed Successfully!
```
As you can see above, I've created options to set **DNS Server IP Address** and **DNS Search Domain** to see if that's what you're looking for. 
You can test it by running:
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/debian-v4.sh)"
```
and using the "Advanced" settings.

**Reply by tteck** [reply]
## 2022-12-03

### Changed

- **All LXC's** 
  - Add options to set DNS Server IP Address and DNS Search Domain (Advanced)

**Comment by wleelavivat** [comment]
Tested. Worked for me. Thanks for a quick update!

**Comment by s-celles** [comment]
I installed both Pi-hole (192.168.1.99) and Traefik (192.168.1.98) with LXC containers but I'm facing an issue.

From my Traefik container I can't ping pi.hole (nor others devices in my network that have a DNS entry in Pi-hole).

My Traefik LXC container have been set with

    # cat /etc/pve/lxc/105.conf
    #<div align='center'><a href='https%3A//Helper-Scripts.com' target='_blank' rel='noopener noreferrer'><img src='https%3A//raw.githubusercontent.com/tteck/Proxmox/main/misc/images/logo-81x112.png'/></a>
    #
    #  # Traefik LXC
    #
    #  <a href='https%3A//ko-fi.com/proxmoxhelperscripts'><img src='https%3A//img.shields.io/badge/&#x2615;-Buy me a coffee-blue' /></a>
    #  </div>
    arch: amd64
    cores: 1
    dns: 192.168.1.99
    features: keyctl=1,nesting=1
    hostname: traefik
    memory: 512
    nameserver: 192.168.1.99
    net0: name=eth0,bridge=vmbr0,hwaddr=BC:24:11:DA:37:74,ip=dhcp,type=veth
    onboot: 1
    ostype: debian
    rootfs: local-lvm:vm-105-disk-0,size=2G
    swap: 512
    tags: proxmox-helper-scripts
    unprivileged: 1

but /etc/resolv.conf in Traefik LXC looks like

    # cat /etc/resolv.conf 
    domain home
    search home
    nameserver 192.168.1.1

(instead of 1.99)

as if Proxmox was overriding this file at container startup.

After manually editing this this from 1.1 to 1.99 I can ping pi.hole
but that's just a temporary fix as rebooting Traefik LXC wipe my changes.

Any idea?
